air/src/passes/translate_from_mir.rs

```diff
@@ -197,8 +223,9 @@ impl AirBuilder<'_> {
     /// Will panic when encountering an unexpected operation
     /// (i.e. that is not a binary operation, a value, enf node or an accessor)
     fn insert_mir_operation(&mut self, mir_node: &Link<Op>) -> Result<NodeIndex, CompileError> {
-        let mir_node = indexed_accessor(mir_node);
+        let mir_node = accessor_to_scalar(mir_node);
         let mir_node = vec_to_scalar(&mir_node);
+        let mir_node = accessor_to_scalar(&mir_node);
         let mir_node_ref = mir_node.borrow();
```

Comment:
I would add some comments to explain what is happening in this sequence

Reply:
@Soulthym - would you be able to address this comment? Or we can merge as is and then @Leo-Besancon could do it in a separate PR (if needed).